node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -1,42 +1,88 @@
 package org.bitcoins.node
 
-import akka.Done
 import akka.actor.ActorSystem
 import org.bitcoins.chain.config.ChainAppConfig
 import org.bitcoins.chain.models.BlockHeaderDAO
 import org.bitcoins.core.api.chain.ChainQueryApi.FilterResponse
 import org.bitcoins.core.protocol.BlockStamp
 import org.bitcoins.node.config.NodeAppConfig
 import org.bitcoins.node.models.Peer
+import org.bitcoins.node.networking.P2PClient
+import org.bitcoins.node.networking.peer._
 
-import scala.concurrent.{Future, Promise}
+import scala.concurrent.Future
 
 case class NeutrinoNode(
-    nodePeer: Peer,
+    private val _nodePeers: Vector[Peer],
+    private val _dataMessageHandler: DataMessageHandler,
     nodeConfig: NodeAppConfig,
     chainConfig: ChainAppConfig,
-    initialSyncDone: Option[Promise[Done]],
     actorSystem: ActorSystem)
     extends Node {
   require(
     nodeConfig.nodeType == NodeType.NeutrinoNode,
     s"We need our Neutrino mode enabled to be able to construct a Neutrino node!")
 
-  implicit override def system: ActorSystem = actorSystem
+  implicit override val system: ActorSystem = actorSystem
 
-  implicit override def nodeAppConfig: NodeAppConfig = nodeConfig
+  implicit override val nodeAppConfig: NodeAppConfig = nodeConfig
 
-  override def chainAppConfig: ChainAppConfig = chainConfig
+  implicit override val chainAppConfig: ChainAppConfig = chainConfig
 
-  override val peer: Peer = nodePeer
+  protected var peers: Vector[Peer] = _nodePeers
```

Comment:
Probably should be a `Set`? Although we do need to make sure our `Peer` equality checks actually work